tests/integration/test_metadata_model.py

```diff
@@ -12,36 +25,112 @@
+            "syn62822337",
+        ),
+    ]
+
     @pytest.mark.parametrize(
-        "manifest_path, dataset_id, validate_component, expected_manifest_id, dataset_scope",
-        [
-            (
-                "mock_manifests/filepath_submission_test_manifest.csv",
-                "syn62276880",
-                None,
-                "syn62280543",
-                None,
-            ),
-            (
-                "mock_manifests/ValidFilenameManifest.csv",
-                "syn62822337",
-                "MockFilename",
-                "syn62822975",
-                "syn62822337",
-            ),
-        ],
+        "manifest_path, dataset_id, validate_component, expected_manifest_id, "
+        "expected_table_id, dataset_scope",
+        test_cases,
     )
-    def test_submit_filebased_manifest(
+    def test_submit_filebased_manifest_file_and_entities(
         self,
         helpers,
         manifest_path,
         dataset_id,
         validate_component,
         expected_manifest_id,
+        expected_table_id,
         dataset_scope,
         mocker: MockerFixture,
+        synapse_store,
     ):
-        # spys
+        self._submit_and_verify_manifest(
+            helpers=helpers,
+            mocker=mocker,
+            synapse_store=synapse_store,
+            manifest_path=manifest_path,
+            dataset_id=dataset_id,
+            expected_manifest_id=expected_manifest_id,
+            expected_table_id=expected_table_id,
+            manifest_record_type="file_and_entities",
+            validate_component=validate_component,
+            dataset_scope=dataset_scope,
+        )
+
+    @pytest.mark.parametrize(
+        "manifest_path, dataset_id, validate_component, expected_manifest_id, "
+        "expected_table_id, dataset_scope",
+        test_cases,
+    )
+    def test_submit_filebased_manifest_table_and_file(
+        self,
+        helpers,
+        manifest_path,
+        dataset_id,
+        validate_component,
+        expected_manifest_id,
+        expected_table_id,
+        dataset_scope,
+        mocker: MockerFixture,
+        synapse_store,
+    ):
+        self._submit_and_verify_manifest(
+            helpers=helpers,
+            mocker=mocker,
+            synapse_store=synapse_store,
+            manifest_path=manifest_path,
+            dataset_id=dataset_id,
+            expected_manifest_id=expected_manifest_id,
+            expected_table_id=expected_table_id,
+            manifest_record_type="table_and_file",
+            validate_component=validate_component,
+            dataset_scope=dataset_scope,
+        )
+
+    def _submit_and_verify_manifest(
+        self,
+        helpers,
+        mocker,
+        synapse_store,
+        manifest_path,
+        dataset_id,
+        expected_manifest_id,
+        expected_table_id,
+        manifest_record_type,
+        validate_component=None,
+        dataset_scope=None,
+        download_dir=os.path.join(os.getcwd(), "temp"),
```

Comment:
we probably shouldn't download them to a fixed directory because what if one workflow
creates this temp directory, and another concurrent workflow attempts to delete it?  What do you think @BryanFauble ?

Reply:
This is a good point @linglp, I've re-worked this logic to use a native python package called `tempfile` instead which builds and tears down a unique directory each time: https://github.com/Sage-Bionetworks/schematic/pull/1485/commits/e3da3d2cf25a6e31e1a17be2d91878d01aa70264

Reply:
This is looking very good Jenny! Thank you again! I appreciate a lot!